hawkular-dmr-client/src/main/java/org/hawkular/dmr/api/OperationBuilder.java

```diff
@@ -0,0 +1,751 @@
+         *
+         * @param name
+         * @param value
+         * @return
+         */
+        @SuppressWarnings("unchecked")
+        public T attribute(String name, Integer value) {
+            if (value != null) {
+                baseNode.get(name).set(value);
+            }
+            return (T) this;
+        }
+
+        /**
+         * If {@code value == null}, no entry is added. Otherwise, the attribute will be added.
+         *
+         * @param name
+         * @param value
+         * @return
+         */
+        @SuppressWarnings("unchecked")
+        public T attribute(String name, String value) {
+            if (value != null) {
+                baseNode.get(name).set(value);
+            }
+            return (T) this;
+        }
+
+        public T valueAttribute(String value) {
+            return attribute(ModelDescriptionConstants.VALUE, value);
+        }
+
+    }
+
+    public static class AddressBuilder<T extends AddressBuilder<?, P>, P extends AbstractSingleOperationBuilder<?, ?>> {
+        private final ModelNode baseNode = new ModelNode();
+        private final P parentBuilder;
+
+        private AddressBuilder(P parentBuilder) {
+            super();
+            this.parentBuilder = parentBuilder;
+        }
+
+        public ModelNode build() {
+            return baseNode;
+        }
+
+        @SuppressWarnings("unchecked")
+        public T datasource(String datasourceName) {
+            baseNode.add(DATASOURCE, datasourceName);
+            return (T) this;
+        }
+
+        public P parentBuilder() {
+            parentBuilder.address(baseNode);
+            return parentBuilder;
+        }
+
+        @SuppressWarnings("unchecked")
+        public T segment(String key, String value) {
+            baseNode.add(key, value);
+            return (T) this;
+        }
+
+        @SuppressWarnings("unchecked")
+        public T segments(ModelNode adr) {
+            for (Property prop : adr.asPropertyList()) {
+                segment(prop.getName(), prop.getValue().asString());
+            }
+            return (T) this;
+        }
+
+        @SuppressWarnings("unchecked")
+        public T segments(String path) {
+            StringTokenizer st = new StringTokenizer(path, "/");
+            while (st.hasMoreTokens()) {
+                String token = st.nextToken();
+                int eqPos = token.indexOf('=');
+                if (eqPos >= 0) {
+                    String key = token.substring(0, eqPos);
+                    String value = token.substring(eqPos + 1);
+                    segment(key, value);
+                } else {
+                    /* ignore */
+                }
+            }
+            return (T) this;
+        }
+
+        public T subsystem(String subsystem) {
+            return segment(ModelDescriptionConstants.SUBSYSTEM, subsystem);
+        }
+
+        public T subsystemDatasources() {
```

Comment:
again, seems odd that only datasource stuff is here. but I guess this is just for convenience since you use ths in the datasource stuff.